src/components/auth/SignUp.jsx

```diff
@@ -0,0 +1,167 @@
+          />
+          <h2 className="mt-0 text-center text-3xl font-bold tracking-tight text-gray-900">
+            Sign up to get started
+          </h2>
+        </div>
+
+        <form className="mt-8 space-y-6" onSubmit={handleSubmit(onSubmit)}>
+          <input type="hidden" name="remember" defaultValue="true" />
+          <div className="-space-y-px rounded-md shadow-sm">
+            <div>
+              <label htmlFor="name" className="sr-only">
+                Name
+              </label>
+              <input
+                type="text"
+                name="name"
+                {...register('name', { required: true })}
+                className={`block w-full rounded-md border-0 py-2 px-3.5 text-gray-900 shadow-sm ring-1 ring-inset ring-gray-300 placeholder:text-gray-400 focus:ring-2 focus:ring-focus-color sm:text-sm sm:leading-6 mb-6 ${
+                  errors.name ? 'border-red-500' : ''
+                }`}
+                placeholder="User name"
+              />
+              <div className="text-red-600">
+                {errors.name && (
+                <small className="text-danger">Name is required</small>
+                )}
+              </div>
+            </div>
+
+            <div>
+              <label htmlFor="email-address" className="sr-only">
+                Email address
+              </label>
+              <input
+                id="email-address"
+                type="email"
+                name="email"
+                {...register('email', {
+                  required: true,
+                  pattern: /^\S+@\S+$/i,
+                })}
+                className={`block w-full rounded-md border-0 py-2 px-3.5 text-gray-900 shadow-sm ring-1 ring-inset ring-gray-300 placeholder:text-gray-400 focus:ring-2 focus:ring-inset focus:ring-focus-color sm:text-sm sm:leading-6 mb-6 ${
+                  errors.email ? 'border-red-500' : ''
+                }`}
+                placeholder="Email address"
+              />
+              <div className="text-red-600">
+                {errors.email && (
+                <small className="text-danger">Invalid email address</small>
+                )}
+              </div>
+            </div>
+            <div>
+              <label htmlFor="password" className="sr-only">
+                Password
+              </label>
+              <input
+                id="password"
+                name="password"
+                type="password"
+                {...register('password', { required: true })}
+                onChange={(e) => setValues({ ...values, [e.target.name]: e.target.value })}
+                className={`block w-full rounded-md border-0 py-2 px-3.5 text-gray-900 shadow-sm ring-1 ring-inset ring-gray-300 placeholder:text-gray-400 focus:ring-2 focus:ring-inset focus:ring-focus-color sm:text-sm sm:leading-6 mb-6 ${
+                  errors.password ? 'border-red-500' : ''
+                }`}
+                placeholder="Password"
```

Comment:
[OPTIONAL] This is a bit clunky. Consider separating form submission logic to its own method